From a087295ba496680b81ecafe72f999d4e9076e737 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Fri, 23 Jun 2023 09:13:39 +0200
Subject: [PATCH] addons: ethtool: add rx-vlan-filter

some nic like mellanox connectx don't work well with
vlan aware bridge && rx-vlan-filter.
(They are limited in number of vlans in hardware filtering,
and break with big number of vlans like bridge-vids 2-4096)

Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
---
 ifupdown2/addons/ethtool.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ifupdown2/addons/ethtool.py b/ifupdown2/addons/ethtool.py
index 2134329..8a79472 100644
--- a/ifupdown2/addons/ethtool.py
+++ b/ifupdown2/addons/ethtool.py
@@ -117,6 +117,12 @@ class ethtool(Addon, moduleBase):
                 'validvals': ['on', 'off'],
                 'default': 'varies by interface'
             },
+            'rx-vlan-filter': {
+                'help': 'RX Vlan Filter',
+                'example': ['rx-vlan-filter off'],
+                'validvals': ['on', 'off'],
+                'default': 'varies by interface'
+            },
             'ring-rx': {
                 'help': 'Ring RX Parameter',
                 'example': ['ring-rx 512'],
@@ -384,6 +390,7 @@ class ethtool(Addon, moduleBase):
         self.do_offload_settings(ifaceobj, 'ufo-offload', 'ufo')
         self.do_offload_settings(ifaceobj, 'tx-offload', 'tx')
         self.do_offload_settings(ifaceobj, 'rx-offload', 'rx')
+        self.do_offload_settings(ifaceobj, 'rx-vlan-filter', 'rx-vlan-filter')
 
     def _pre_down(self, ifaceobj):
         pass #self._post_up(ifaceobj,operation="_pre_down")
-- 
2.39.2

